test/sol6/multipleFeeHandlers.js

```diff
@@ -0,0 +1,625 @@
+            assertBalancesIncreased(initialBalance, newBalance, "staker's token balance did not increase"); 
+            initialBalance = newBalance;
+
+            // should claim only from tempUsdcFeeHandler1
+            selectedEpoch = tempUsdcFeeHandler1StartEpoch + 2;
+            Helper.assertEqual(await tempUsdcFeeHandler2.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await tempUsdcFeeHandler1.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await usdcFeeHandler.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            await feeWrapper.claimStakerReward(staker, selectedEpoch, tokenIndex, tokenIndex + 1, 0, 5);
+            Helper.assertEqual(await tempUsdcFeeHandler2.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await tempUsdcFeeHandler1.hasClaimedReward(staker, selectedEpoch), true, "staker should have claimed reward");
+            Helper.assertEqual(await usdcFeeHandler.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            newBalance = await getBalances(staker, [usdc]);
+            assertBalancesIncreased(initialBalance, newBalance, "staker's token balance did not increase"); 
+            initialBalance = newBalance;
+
+            // should claim only from usdcFeeHandler
+            selectedEpoch = tempUsdcFeeHandler1StartEpoch - 1;
+            Helper.assertEqual(await tempUsdcFeeHandler2.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await tempUsdcFeeHandler1.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await usdcFeeHandler.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            await feeWrapper.claimStakerReward(staker, selectedEpoch, tokenIndex, tokenIndex + 1, 0, 5);
+            Helper.assertEqual(await tempUsdcFeeHandler2.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await tempUsdcFeeHandler1.hasClaimedReward(staker, selectedEpoch), false, "staker should not have claimed reward");
+            Helper.assertEqual(await usdcFeeHandler.hasClaimedReward(staker, selectedEpoch), true, "staker should have claimed reward");
+            newBalance = await getBalances(staker, [usdc]);
+            assertBalancesIncreased(initialBalance, newBalance, "staker's token balance did not increase"); 
+            initialBalance = newBalance;
+        });
+
+        it("should claim from multiple feeHandlers if for overlapping epochs", async() => {
+            // should claim from both tempUsdcFeeHandler2 and tempUsdcFeeHandler1
```

Comment:
I don't understand over lapping epochs.
each fee handler has only start epoch. and they have to bigger each time fee handler is added. right?

Reply:
Not strictly bigger. It's possible to add (update) N feeHandlers in 1 epoch and hence 1 or more feeHandlers to have the same starting epoch.

Reply:
OK
Got it